### ISSUE TITLE
Can't provisioning a v1.24.7-rc1+rke2r1 cluster with Rancher

### ISSUE THREAD

**Issue by bk201** [rancher/rke2#3464]
**Environmental Info:**
RKE2 Version: [v1.24.7-rc1+rke2r1](https://github.com/rancher/rke2/releases/tag/v1.24.7-rc1%2Brke2r1)

Node(s) CPU architecture, OS, and Version: 

Cluster Configuration:

**Describe the bug:**

We see Rancher (v2.6.9-rc5) can't provision cluster when switching to `v1.24.7-rc1+rke2r1`.

The cluster is in provisioning:
```
$ kubectl get clusters.provisioning.cattle.io local -n fleet-local -o yaml
...
  - lastUpdateTime: "2022-10-17T03:54:38Z"
    message: non-ready bootstrap machine(s) custom-b7f58c72a37e and join url to be
      available on bootstrap node
    reason: Waiting
    status: Unknown
    type: Updated
```

By checking Rancher logs, we observed these lines repeat:
```
2022/10/17 04:51:16 [ERROR] [CAPI] Reconciler error: providerID must be of the form <cloudProvider>://<optional>/<segments>/<provider id>
2022/10/17 04:54:29 [INFO] [planner] rkecluster fleet-local/local: waiting: non-ready bootstrap machine(s) custom-b7f58c72a37e and join url to be available on bootstrap node
2022/10/17 05:04:30 [INFO] [planner] rkecluster fleet-local/local: waiting: non-ready bootstrap machine(s) custom-b7f58c72a37e and join url to be available on bootstrap node
```

The machine manifest (in v1.24.6-rke2r1, the providerID looks like: `rke2://node1`):
```
$ kubectl get machine custom-b7f58c72a37e -n fleet-local -o yaml
...
spec:
  bootstrap:
    configRef:
      apiVersion: rke.cattle.io/v1
      kind: RKEBootstrap
      name: custom-b7f58c72a37e
      namespace: fleet-local
    dataSecretName: custom-b7f58c72a37e-machine-bootstrap
  clusterName: local
  infrastructureRef:
    apiVersion: rke.cattle.io/v1
    kind: CustomMachine
    name: custom-b7f58c72a37e
    namespace: fleet-local
  providerID: rke2
```

**Steps To Reproduce:**

For a quick reproduce:
- Clone https://github.com/bk201/vagrant-rancherd
- Edit settings.yaml to
```
kubernetes_version: v1.24.7-rc1+rke2r1
rancher_version: v2.6.9-rc5
```
- `vagrant up node1`
- `./kube-env.sh` to get a valid kubeconfig.

**Expected behavior:**

Providing a valid providerID so the cluster can provision.

**Actual behavior:**

**Additional context / logs:**

**Comment by brandond** [comment]
This is probably related to https://github.com/k3s-io/k3s/commit/063039471c9fa0197a3bb49899c3d4cf462ffc68.

Unfortunately we don't have any tests that cover the providerId field on the k3s or rke2 side as we don't use them for anything and were not aware that they were consumed my anything downstream.

**Comment by bk201** [comment]
@brandond We can confirm the issue is fixed in v1.24.7-rc2. Now the providerID is correct. Thanks!
